template/example_test.go

```diff
@@ -50,44 +50,51 @@ func ExampleTemplate_complexMessage() {
 .input { $color :color style=RGB}
 {{John is { $age } years old and his favorite color is { $color }.}}`
 
-	color := func(value any, options template.Options, locale language.Tag) (any, error) {
+	color := func(value any, options template.Options, locale language.Tag) (*template.ResolvedValue, error) {
+		errorf := func(format string, args ...any) (*template.ResolvedValue, error) {
+			return template.NewResolvedValue(""), fmt.Errorf("exec color function: "+format, args...)
```

Comment:
Does this still work? Do registry func consumers care about returned value on error?

```suggestion
			return nil, fmt.Errorf("exec color function: "+format, args...)
```

Reply:
Good point, replaced.